redash/handlers/query_results.py

```diff
@@ -96,6 +96,15 @@ def run_query(query, parameters, data_source, query_id, max_age=0):
         return {'job': job.to_dict()}
 
 
+def get_download_filename(query_result, query=None, filetype='json'):
+    str_date = query_result.retrieved_at.strftime("%Y-%m-%d")
+    if query is not None:
+        str_filename = to_filename(query.name) if query.name != '' else str(query.id)
+    else:
+        str_filename = str(query_result.id)
+    return u"{}.{}.{}".format(str_filename, str_date, filetype)
```

Comment:
Isn't it better to use underscores (or sth else) instead of dots to separate the filename? Windows may have issues with it, not sure :thinking:

Reply:
@gabrieldutra Windows doesn't have issues with dots since time when started using Fat32 :slightly_smiling_face:  Also, Windows started using Unicode for filenames approx. from the same time - I think it was Win95 or Win98.

Reply:
Well I'm still using Norton Commander and this is really not ok

Reply:
@ranbena NC was developed for MS DOS, and it's miracle that it worked in XP/Vista/7. FAR Manager is a modern replacement for NC, and it correctly works with Unicode filenames (I was using it few years ago on Win7)

Reply:
I also consider this a matter of standard, a dot is commonly used to separate filename and extension. Windows can work fine with it today, but what about programs?

Reply:
@gabrieldutra All programs that were developed at times of WinXP can work with it smoothly. Some very old apps may not handle it properly, but, well, they cannot handle extensions like `.json` or `.xlsx` (more that 3 characters) as well. I think we shouldn't focus on so old software.

Reply:
BTW - WinXP (an newer versions) have a fallback for such old applications - DOS-alike filenames that exists simultaneously to regular ones: https://docs.microsoft.com/en-us/windows/desktop/FileIO/naming-a-file#short-vs-long-names

Reply:
I'm with @gabrieldutra: I think it's better to use underscores. It's also consistent with what we were doing until now. Also we were replacing spaces with underscores (in query name).

Reply:
I was actually thinking about recent programs/scripts that could have issues with it by treating this  naively in the code. Personally, I've always tried to avoid it. Anyway, as this seems to be a really small context nowadays, I won't make big deal with it :slightly_smiling_face:

Reply:
@arikfr But what if query name itself will contain a dot?

Reply:
🤷‍♂️ 

Anyway, the dot vs. underscore isn't a big deal, but I do prefer we replace whitespace with underscores (makes it easier to use with CLI tools).

Reply:
@arikfr Done.